### PR TITLE
Bug fix in Semantic Segmentation + enable DINOV2 export in ONNX

### PR DESCRIPTION
**Pull request by kprokofi**
### Summary
1) Only without registers DINOV2 can be exported to ONNX. The accuracy difference is insignificant
2) Bug fix with ignore index in labels pre-filtering. Bug fix with overriding default values in segmentation models

### Checklist

- [ ] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [ ] I have ran e2e tests and there is no issues.
- [ ] I have added the description of my changes into CHANGELOG in my target branch (e.g., [CHANGELOG](https://github.com/openvinotoolkit/training_extensions/blob/develop/CHANGELOG.md) in develop).​
- [ ] I have updated the documentation in my target branch accordingly (e.g., [documentation](https://github.com/openvinotoolkit/training_extensions/tree/develop/docs) in develop).
- [ ] I have [linked related issues](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

### License

- [ ] I submit _my code changes_ under the same [Apache License](https://github.com/openvinotoolkit/training_extensions/blob/develop/LICENSE) that covers the project.
      Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2024 Intel Corporation
# SPDX-License-Identifier: Apache-2.0
```

### CODE REVIEW

File: src/otx/core/data/pre_filtering.py [hunk 1]
```diff
@@ -16,7 +16,12 @@
     from datumaro.components.dataset_base import DatasetItem
 
 
-def pre_filtering(dataset: DmDataset, data_format: str, unannotated_items_ratio: float) -> DmDataset:
+def pre_filtering(
+    dataset: DmDataset,
+    data_format: str,
+    unannotated_items_ratio: float,
+    ignore_index: int | None = None,
```
Comment (eunwoosh) [inline src/otx/core/data/pre_filtering.py:23]: Could you add docstring of this argument?

File: src/otx/algo/segmentation/dino_v2_seg.py [hunk 2]
```diff
@@ -46,9 +46,8 @@ class OTXDinoV2Seg(TorchVisionCompatibleModel):
 
     def _create_model(self) -> nn.Module:
         # merge configurations with defaults overriding them
-        backbone_configuration = self.backbone_configuration | DinoV2Seg.default_backbone_configuration
-        decode_head_configuration = self.decode_head_configuration | DinoV2Seg.default_decode_head_configuration
-        # initialize backbones
+        backbone_configuration = DinoV2Seg.default_backbone_configuration | self.backbone_configuration
+        decode_head_configuration = DinoV2Seg.default_decode_head_configuration | self.decode_head_configuration
```
Comment (harimkang) [inline src/otx/algo/segmentation/dino_v2_seg.py:50]: Why is `self.**`_configuration separate, and is this something that can actually be used?

Comment (eunwoosh) [reply src/otx/algo/segmentation/dino_v2_seg.py:50]: a_dict | b_dict is same as a_dict.update(b_dict). So, self.**_configuration has a priority than default configuration.

**Review by eunwoosh — APPROVED** [review]

**Review by harimkang — APPROVED** [review]

File: src/otx/algo/segmentation/dino_v2_seg.py (hunk 2 above)
Comment (kprokofi) [reply src/otx/algo/segmentation/dino_v2_seg.py:50]: I added this default configuration for simple API usage. Let's imagine that we don't have it. Then user should create model and pass all parameters to initialize backbone and head. It is obvious overhead. Now, he can provide only few parameters to override default values.
self.** is what comes from config or what is passed to model class.
